File: DS_P1_S3/C++/GestorFiltros.cpp
#include "GestorFiltros.h"

GestorFiltros::GestorFiltros() {
    cadena = new CadenaFiltros();
}

void GestorFiltros::setFiltros(Filtro* filtro) {
    cadena->addFiltro(filtro);
}

FormularioPrestamo* GestorFiltros::peticionFiltros(FormularioPrestamo* form) {
    return cadena->ejecutar(form);
}

Prestamo GestorFiltros::pedirPrestamo(FormularioPrestamo* f) {
    f = peticionFiltros(f);
    return cadena->pedirPrestamo(f);
}

File: DS_P1_S3/Dart/ds_p1_s3/lib/GestorFiltros.dart
import 'package:ds_p1_s3/CadenaFiltros.dart';
import 'package:ds_p1_s3/Filtro.dart';
import 'package:ds_p1_s3/prestamo.dart';

import 'FormularioPrestamo.dart';

class GestorFiltros{
  CadenaFiltros cadena;

  GestorFiltros(){
    cadena = new CadenaFiltros();
  }
  Prestamo pedirPrestamo(FormularioPrestamo f){
    f = peticionFiltros(f);
    return cadena.pedirPrestamo(f);
  }
  peticionFiltros(FormularioPrestamo f){
    return cadena.ejecutar(f);
  }

  void setFiltros(Filtro f){
    cadena.addFiltro(f);
  }
}